addons/website/models/website_page.py

```diff
@@ -209,8 +142,40 @@ def unlink(self):
         return super(Page, self).unlink()
 
     def write(self, vals):
-        if 'url' in vals and not vals['url'].startswith('/'):
-            vals['url'] = '/' + vals['url']
+        for page in self:
+            website_id = False
+            if vals.get('website_id') or page.website_id:
+                website_id = vals.get('website_id') or page.website_id.id
+
+            # If URL has been edited, slug it
+            if 'url' in vals:
+                url = vals['url']
+                redirect_old_url = redirect_type = None
+                if isinstance(url, dict):
+                    redirect_old_url = url.get('redirect_old_url')
+                    redirect_type = url.get('redirect_type')
+                    url = url.get('url')
```

Comment:
That's quite hacky to say the least..
Shouldn't we then go with a wizard instead of a form view?